### PR TITLE
+docs #218 javadsl implementation of completeWith & handleWith directives

### PR DESCRIPTION
**Pull request by jedrekn**
Added implementation of javadsl's version of handleWith and completeWith directives (javadsl's entity directive was used for implementation reference and MarshallingDirectivesExamplesSpec for tests reference). Documentation updated to point to newly created example specs.

### CODE REVIEW

**Comment by typesafehub-validator** [comment]
Hi @jedrekn,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Lightbend Contributors License Agreement:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

**Comment by jedrekn** [comment]
Refs #218

**Comment by jedrekn** [comment]
I've just signed the CLA.

**Comment by ktoso** [comment]
OK TO TEST

File: docs/src/test/java/docs/http/javadsl/server/directives/MarshallingDirectivesExamplesTest.java [hunk 1]
```diff
@@ -58,4 +63,58 @@ public void testEntity() {
...
   }
+
+  @Test
+  public void testCompleteWith() {
+    //#example-completeWith-with-json
+    final Marshaller<Person, HttpResponse> marshaller = Marshaller.entityToOKResponse(Jackson.<Person>marshaller());
+
+    final Consumer<Consumer<Person>> findPerson = completionFunction -> {
+
+      //... some processing logic...
```
Comment (jrudolph) [inline docs/src/test/java/docs/http/javadsl/server/directives/MarshallingDirectivesExamplesTest.java]: You could even pass the `completionFunction` to a different thread here. Maybe mention that in a comment as well?

**Comment by jrudolph** [comment]
PLS BUILD

File: docs/src/test/java/docs/http/javadsl/server/directives/MarshallingDirectivesExamplesTest.java (hunk 1 above)
Comment (jedrekn) [inline docs/src/test/java/docs/http/javadsl/server/directives/MarshallingDirectivesExamplesTest.java]: Sorry didn't see your comment before. Will add it.

**Comment by jrudolph** [comment]
LGTM.

**Comment by jrudolph** [comment]
PLS BUILD

**Comment by jrudolph** [comment]
Thanks again, @jedrekn. Nice change.

**Pull request merged**